demos/mqtt/mqtt_demo_mutual_auth/mqtt_demo_mutual_auth.c

```diff
@@ -1147,11 +1166,22 @@ static int establishMqttSession( MQTTContext_t * pMqttContext,
      * the metrics string is appended to the username to support both client
      * authentication and metrics collection. */
     #ifdef CLIENT_USERNAME
-        connectInfo.pUserName = CLIENT_USERNAME_WITH_METRICS;
-        connectInfo.userNameLength = strlen( CLIENT_USERNAME_WITH_METRICS );
+        pMemchrPtr = memchr( CLIENT_USERNAME, '?', strlen( CLIENT_USERNAME ) );
+
+        if( pMemchrPtr != NULL )
+        {
+            connectInfo.pUserName = CLIENT_USERNAME_APPEND_METRICS;
+            connectInfo.userNameLength = strlen( CLIENT_USERNAME_APPEND_METRICS );
+        }
+        else
+        {
+            connectInfo.pUserName = CLIENT_USERNAME_WITH_METRICS;
```

Comment:
Is it allowed for users to provide username without '?' ?

Reply:
If it's not allowed, can we just do error handling in this case and delete `CLIENT_USERNAME_WITH_METRICS`?

Reply:
It is allowed for user to provide username.